src/components/AlarmsTable.tsx

```diff
@@ -48,8 +49,19 @@ function WarningAreaRow({ warningArea, alarm, now, operationalModelLevel }: RowP
   const maxForecast = useMaxForecastAtPoint(operationalModelLevel, alarm || null);
 
   const warningLevel = alarm ? alarm.latest_trigger.warning_level : null;
-  const warningClass = warningLevel ? styles[`tr_${warningLevel.toLowerCase()}`] : "";
-  const warningClassTd = warningLevel ? styles.td_warning : "";
+  const latestEWNRssItem = useLatestItemForArea(warningArea.properties.name);
+  const warningClassTd = warningLevel
+    ? styles[`td_${warningLevel.toLowerCase()}`] + " " + styles.td_warning
+    : "";
+
+  let EWNWarning = "-";
+  let warningClassTdEwn = "";
+  if (latestEWNRssItem !== null && latestEWNRssItem.warning.toLowerCase() !== "no further impact") {
+    EWNWarning = latestEWNRssItem.warning;
+    // Turn a warning string like "Minor Flood Warning" into the "td_warning" class
+    warningClassTdEwn = styles[`td_${EWNWarning.toLowerCase().split(" ")[0]}`];
+  }
+
```

Comment:
I would prefer explicit if else or case statements with hardcoded classnames instead of generating them based on a text input. The classes in the css stylesheet are hardcoded anyway so this just adds complexity imo.

Reply:
Will fix